src/tools/pixelate/pixelatetool.cpp

```diff
@@ -35,27 +35,22 @@ QString PixelateTool::description() const
 
 CaptureTool* PixelateTool::copy(QObject* parent)
 {
-    return new PixelateTool(parent);
+    auto* tool = new PixelateTool(parent);
+    copyParams(this, tool);
+    return tool;
 }
 
-void PixelateTool::process(QPainter& painter,
-                           const QPixmap& pixmap,
-                           bool recordUndo)
+void PixelateTool::process(QPainter& painter, const QPixmap& pixmap)
 {
-
-    if (recordUndo) {
-        updateBackup(pixmap);
-    }
-
-    QPoint& p0 = m_points.first;
-    QPoint& p1 = m_points.second;
+    QPoint p0 = points().first;
+    QPoint p1 = points().second;
     QRect selection = QRect(p0, p1).normalized();
     auto pixelRatio = pixmap.devicePixelRatio();
     QRect selectionScaled =
       QRect(p0 * pixelRatio, p1 * pixelRatio).normalized();
 
-    // If thickness is less than 1, use old blur process
-    if (m_thickness <= 1) {
+    // If thickness is 0, use old blur process
+    if (thickness() <= 0) {
```

Comment:
I think this is a bug because thickness() never returns a value of 0, therefore the old blur cannot be used.

Reply:
Thickness can be zero, but I'll check it. Possibly I made a new error with thickness somewhere.